source/lib.ts

```diff
@@ -0,0 +1,264 @@
+// /source/lib.ts
+// The option parser and rate limiting middleware
+
+import Express from 'express'
+
+import { Options, AugmentedRequest, RateLimitRequestHandler } from './types.js'
+import MemoryStore from './memory-store.js'
+
+/**
+ * Adds the defaults for options the user has not specified.
+ *
+ * @param options {Options} - The options the user specifies
+ *
+ * @returns {Options} - A complete configuration object
+ */
+const parseOptions = (passedOptions: Partial<Options>): Options => {
+	// Now add the defaults for the other options
+	const options: Options = {
+		windowMs: 60 * 1000,
+		store: new MemoryStore(passedOptions.windowMs ?? 60 * 1000),
+		max: 5,
+		message: 'Too many requests, please try again later.',
+		statusCode: 429,
+		headers: true,
+		useStandardizedHeaders: false,
```

Comment:
Ditto above comments about renaming these. I think I like `legacyHeaders` and `standardHeaders` - but I think we should continue defaulting to legacy on and standard off for backwards compatibility. (As you've done here.)